kit/src/elements/textarea/mod.rs

```diff
@@ -126,13 +143,24 @@ pub fn Input<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
                         let shift_key_as_modifier = evt.data.modifiers().contains(Modifiers::SHIFT);
 
                         if enter_pressed && !shift_key_as_modifier {
+                            if *show_char_counter {
```

Comment:
if the space key increments the char counter, then enter + shift, which creates a newline, should probably also increment the char counter.